### PR TITLE
Cut 2.0.0 rc.1

### PR DESCRIPTION
**Pull request by mrueg**
* [CHANGE] Rename --labels-metric-allow-list to --metric-labels-allowlist #1424
* [CHANGE] Remove deprecated Kubernetes APIs #1423
* [CHANGE] Remove vendor folder #1419

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## v2.0.0-rc.1 / 2021-03-26
+
+* [CHANGE] Rename --labels-metric-allow-list to --metric-labels-allowlist #1424
+* [CHANGE] Remove deprecated Kubernetes APIs #1423
+* [CHANGE] Remove vendor folder #1419
+
```
Comment (lilic) [inline CHANGELOG.md:6]: I would include the https://github.com/kubernetes/kube-state-metrics/commit/3ab8b19c29a0008e93171a11842cfe8a2c558686 as some of our users use kube-state-metrics as library, this should be helpful to them.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+## v2.0.0-rc.1 / 2021-03-26
+
+* [CHANGE] Rename --labels-metric-allow-list to --metric-labels-allowlist #1424
```
Comment (lilic) [inline CHANGELOG.md:3]: We also want to stop creating quay images, not sure if this is the right place to mention this or in the release itself, or both.

See https://github.com/kubernetes/kube-state-metrics/issues/1409 for background.

**Review by lilic — COMMENTED** [review]
Thank you! 🎉 
Couple of suggestions only.

File: CHANGELOG.md (hunk 1 above)
Comment (mrueg) [reply CHANGELOG.md:9]: Thanks, I added a change log.

File: CHANGELOG.md (hunk 2 above)
Comment (mrueg) [reply CHANGELOG.md:3]: I added a note about quay.io and k8s.gcr.io

**Review by lilic — APPROVED** [review]
/lgtm

Thanks! 🚀

**Pull request merged**